sdk/appcenter-identity/src/test/java/com/microsoft/appcenter/identity/IdentityTest.java

```diff
@@ -276,6 +277,63 @@ public void signInThenDownloadValidConfiguration() throws Exception {
         verifyStatic();
     }
 
+    @Test
+    public void disableServiceDuringDownloadingConfiguration() throws Exception {
+
+        /* Mock config call. */
+        ServiceCall getConfigCall = mock(ServiceCall.class);
+        when(mHttpClient.callAsync(anyString(), anyString(), anyMapOf(String.class, String.class), any(HttpClient.CallTemplate.class),
+                any(ServiceCallback.class))).thenReturn(getConfigCall);
+
+        /* Mock JSON. */
+        JSONObject jsonConfig = mockValidForAppCenterConfig();
+
+        /* Mock authentication lib. */
+        PublicClientApplication publicClientApplication = mock(PublicClientApplication.class);
+        whenNew(PublicClientApplication.class).withAnyArguments().thenReturn(publicClientApplication);
+
+        /* Start identity service. */
+        Identity identity = Identity.getInstance();
+        start(identity);
+
+        /* Mock foreground then background again. */
+        identity.onActivityResumed(mock(Activity.class));
+
+        /* Sign in, will be delayed until configuration ready. */
+        Identity.signIn();
```

Comment:
We don't have delayed sign in anymore, comment is probably wrong.

Reply:
<img width="842" alt="Screen Shot 2019-04-19 at 11 28 34 AM" src="https://user-images.githubusercontent.com/623672/56437997-4c865800-6296-11e9-93dd-056a64b95a83.png">

Thus this test setup seems unreal to me, can you fix the semantics of the test?

Reply:
Fixing some tests right now that were not updated after the error on background/not configured change.